go-controller/pkg/testing/libovsdb/libovsdb.go

```diff
@@ -0,0 +1,220 @@
+				s.Close()
+				return
+			}
+		}
+	}()
+
+	c, err := newClient(cfg, clientData, stopChan)
+	if err != nil {
+		close(internalStopChan)
+	}
+
+	return c, err
+}
+
+func newNBClient(cfg config.OvnAuthConfig, clientData LibOVSDBClientData, stopChan chan struct{}) (libovsdbClient.Client, error) {
+	onConnectFn := func(client libovsdbClient.Client) error {
+		return libOVSDBClientOnConnect(client, clientData)
+	}
+	libovsdbOvnNBClient, err := util.NewNBClientWithConfig(cfg, onConnectFn, stopChan)
+	if err != nil {
+		return nil, err
+	}
+	return libovsdbOvnNBClient, err
+}
+
+func newSBClient(cfg config.OvnAuthConfig, clientData LibOVSDBClientData, stopChan chan struct{}) (libovsdbClient.Client, error) {
+	onConnectFn := func(client libovsdbClient.Client) error {
+		return libOVSDBClientOnConnect(client, clientData)
+	}
+	libovsdbOvnSBClient, err := util.NewSBClientWithConfig(cfg, onConnectFn, stopChan)
+	if err != nil {
+		return nil, err
+	}
+	return libovsdbOvnSBClient, err
+}
+
+func newSBServer(cfg config.OvnAuthConfig, data []LibOVSDBServerData) (*server.OvsdbServer, error) {
+	dbModel, err := sbdb.FullDatabaseModel()
+	if err != nil {
+		return nil, err
+	}
+	schema := sbdb.Schema()
+	return newOVSDBServer(cfg, dbModel, schema, data)
+}
+
+func newNBServer(cfg config.OvnAuthConfig, data []LibOVSDBServerData) (*server.OvsdbServer, error) {
+	dbModel, err := nbdb.FullDatabaseModel()
+	if err != nil {
+		return nil, err
+	}
+	schema := nbdb.Schema()
+	return newOVSDBServer(cfg, dbModel, schema, data)
+}
+
+func newOVSDBServer(cfg config.OvnAuthConfig, dbModel *model.DBModel, schema ovsdb.DatabaseSchema, data []LibOVSDBServerData) (*server.OvsdbServer, error) {
+	db := server.NewInMemoryDatabase(map[string]*model.DBModel{
+		schema.Name: dbModel,
+	})
+	s, err := server.NewOvsdbServer(db, server.DatabaseModel{
+		Model:  dbModel,
+		Schema: &schema,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(data) > 0 {
+		m := mapper.NewMapper(&schema)
+		for _, d := range data {
+			row, err := m.NewRow(d.TableName, d.Data)
+			if err != nil {
+				return nil, err
+			}
+			res, _ := db.Insert(d.DBName, d.TableName, uuid.NewString(), row)
+			if res.Error != "" {
+				return nil, fmt.Errorf("%s: %s", res.Error, res.Details)
+			}
+		}
+	}
+
+	sockPath := strings.TrimPrefix(cfg.Address, "unix:")
+	lockPath := fmt.Sprintf("%s.lock", sockPath)
+	fileMutex, err := filemutex.New(lockPath)
+	if err != nil {
+		return nil, err
+	}
+
+	err = fileMutex.Lock()
+	if err != nil {
+		return nil, err
+	}
+	go func() {
+		defer fileMutex.Close()
+		if err := s.Serve(string(cfg.Scheme), sockPath); err != nil {
+			log.Fatalf("libovsdb test harness error: %v", err)
+		}
+	}()
+
+	err = wait.Poll(100*time.Millisecond, 500*time.Millisecond, func() (bool, error) { return s.Ready(), nil })
+	if err != nil {
+		s.Close()
+		return nil, err
+	}
+
+	return s, nil
+}
+
+func libOVSDBClientOnConnect(client libovsdbClient.Client, clientData LibOVSDBClientData) error {
```

Comment:
why can't we re-use the `util` functions for creating the clients in the `FakeOVN`? This would make sure that the behaviour is identical, otherwise we'll have to keep the contents of this function in sync with the util one etc...